File: front-end/src/app/service-models/app/v1/client/s-q-l-report-data-model-s-m.ts
import { SQLReportCellSM } from './s-q-l-report-cell-s-m';
import { SQLReportRowsSM } from './s-q-l-report-rows-s-m';

export class SQLReportDataModelSM {
    clientCompanyDetailId?: number;
    pageNo!: number;
    pageSize!: number;
    reportDataColumns!: Array<SQLReportCellSM>;
    reportDataRows!: Array<SQLReportRowsSM>;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/SQLReportDataModelSM.cs
using Siffrum.Web.Payroll.ServiceModels.Enums;
using System.Collections.Generic;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class SQLReportDataModelSM
    {
        public int? ClientCompanyDetailId { get; set; }
        public int PageNo { get; set; }
        public int PageSize { get; set; }
        public List<SQLReportCellSM> ReportDataColumns { get; set; }
        public List<SQLReportRowsSM> ReportDataRows { get; set; }
    }
    public class SQLReportCellSM
    {
        public int CellColumnIndex { get; set; }
        public string CellColumnName { get; set; }
        public CellDataType CellDataType { get; set; }
        public object CellValue { get; set; }
    }
    public class SQLReportRowsSM
    {
        public List<SQLReportCellSM> ReportDataCells { get; set; }

    }
}
